File: kube/src/main/java/com/jacksonrakena/infrastructure/util/ContainerUtils.kt
package com.jacksonrakena.infrastructure.util

import org.cdk8s.Size
import org.cdk8s.plus28.ContainerProps
import org.cdk8s.plus28.ContainerResources
import org.cdk8s.plus28.ContainerSecurityContextProps
import org.cdk8s.plus28.Cpu
import org.cdk8s.plus28.CpuResources
import org.cdk8s.plus28.MemoryResources
import kotlin.time.Duration

fun ContainerProps.Builder.applyCommonConfiguration(): ContainerProps.Builder {
    return this.resources(
        ContainerResources.builder()
            .cpu(CpuResources.builder().build())
            .memory(MemoryResources.builder().request(Size.mebibytes(500)).build())
            .build()
    )
        .securityContext(
            ContainerSecurityContextProps.builder().readOnlyRootFilesystem(false).ensureNonRoot(false).build()
        )
}

File: src/util/container-utils.ts
import * as kplus from "cdk8s-plus-34";
import { Size } from "cdk8s";

export function applyCommonConfiguration(
  props: kplus.ContainerProps
): kplus.ContainerProps {
  return {
    ...props,
    resources: {
      cpu: {},
      memory: { request: Size.mebibytes(500) },
    },
    securityContext: {
      readOnlyRootFilesystem: false,
      ensureNonRoot: false,
    },
  };
}
